src/components/InputDropdown/index.tsx

```diff
@@ -0,0 +1,353 @@
+'use client';
+
+import {
+  useState,
+  useEffect,
+  useRef,
+  useMemo,
+  KeyboardEvent,
+  Dispatch,
+  SetStateAction,
+} from 'react';
+import {
+  DropdownErrorText,
+  DropdownInput,
+  DropdownInputContainer,
+  DropdownInputLabel,
+  DropdownInputTag,
+  TagContainer,
+} from './styles';
+import DropdownMenu from '../DropdownMenu';
+import { DropdownContainer } from '../FilterDropdown/styles';
+
+interface CommonProps {
+  placeholder?: string;
+  error?: string;
+  disabled?: boolean;
+  label: string;
+  options: Set<string>;
+}
+
+interface MultiSelectProps extends CommonProps {
+  multi: true;
+  value: Set<string>;
+  setValue: Dispatch<SetStateAction<Set<string>>>;
+}
+
+interface SingleSelectProps extends CommonProps {
+  multi?: false;
+  value: string;
+  setValue: Dispatch<SetStateAction<string>>;
+}
+
+type InputDropdownProps = MultiSelectProps | SingleSelectProps;
+
+export default function InputDropdown({
+  placeholder,
+  error = '',
+  disabled = false,
+  label,
+  options,
+  multi,
+  value,
+  setValue,
+}: InputDropdownProps) {
+  const containerRef = useRef<HTMLDivElement>(null);
+  const inputRef = useRef<HTMLParagraphElement>(null);
+  const itemsRef = useRef<Map<string, HTMLParagraphElement | null>>(new Map());
+  const [focusIndex, setFocusIndex] = useState(-1);
+  const [menuVisible, setMenuVisible] = useState(false);
+  const [inputValue, setInputValue] = useState('');
+
+  // store array format for focus indexing
+  const optionsArray = useMemo(
+    () =>
+      Array.from(new Set(options)).filter(o =>
+        o.toLowerCase().includes(inputValue),
+      ),
+    [options, inputValue],
+  );
+
+  // detect clicking outside of menu box
+  useEffect(() => {
+    function globalClickEvent(this: Document, e: MouseEvent) {
+      if (
+        containerRef.current &&
+        containerRef.current.contains(e.target as Node)
+      )
+        return;
+      setMenuVisible(false);
+    }
+
+    document.addEventListener('click', globalClickEvent);
+
+    return () => {
+      document.removeEventListener('click', globalClickEvent);
+    };
+  }, []);
+
+  // handle when a tag gets clicked
+  function handleRemoveOption(option: string) {
+    if (containerRef.current) containerRef.current.blur();
+    if (inputRef.current) inputRef.current.blur();
+
+    if (multi) {
+      if (!value.has(option)) return;
+
+      const copy = new Set(value);
+      copy.delete(option);
+      setValue(copy);
+    } else {
+      setValue('');
+    }
+  }
+
+  // decide add/remove options
+  function handleSelectOption(option: string) {
+    if (inputRef.current) inputRef.current.innerHTML = '';
+
+    if (multi) {
+      const copy = new Set(value);
+
+      if (value.has(option)) copy.delete(option);
+      else copy.add(option);
```

Comment:
not sure if this is where we would handle this, but: when the tags overflow to the next line, it seems like the cursor moves to the line after:

<img width="1512" alt="Screenshot 2023-11-18 at 6 17 58 PM" src="https://github.com/calblueprint/immigration-justice-project/assets/47801900/7c4ca014-7c7b-4552-9853-0a2345138f5d">

is there some way to change this so the cursor stays on the same line?

Reply:
i'll try